processing/src/main/java/org/apache/druid/math/expr/ApplyFunction.java

```diff
@@ -476,6 +476,44 @@ private <T> Stream<T> filter(T[] array, LambdaExpr expr, SettableLambdaBinding b
     }
   }
 
+  /**
+   * Extended version of {@link FilterFunction} to return a null expr if filtered result turns out to be empty
```

Comment:
hmm, reading the docs for `filter`, https://github.com/apache/druid/blob/master/docs/querying/math-expr.md#apply-functions, it actually claims that it returns null if no elements match. Given that, I suppose we could just fix `filter` to make it match the docs instead of creating a new function. Is it actually useful to have a `filter` that returns empty array if no matches?

Alternatively, if we do decide we want both behaviors, we should update those docs and add new docs for this function.